### PR TITLE
Test Subscription, PaymentMethod and Lease apis

### PR DESCRIPTION
**Pull request by sadlil**

### CODE REVIEW

File: pkg/billing/subscription_test.go [hunk 1]
```diff
@@ -101,17 +103,173 @@ func (s *SubscriptionTestSuite) TestSubscriptionBuy() {
...
+}
+
+func (s *SubscriptionTestSuite) teardownTestSubscriptionChange() {
+	defer s.teardownTestSubscriptionList()
+
 }
 
 func (s *SubscriptionTestSuite) setupTestSubscriptionBuy() {
-	dataBytes, err := ioutil.ReadFile(os.ExpandEnv("$GOPATH") + "/src/github.com/cowrypay/dealer/hack/testdata/product.json")
+	dataBytes, err := ioutil.ReadFile(os.ExpandEnv("$GOPATH") + "/src/github.com/cowrypay/dealer/hack/testdata/per-seat.json")
```
Comment (TamalSaha) [inline pkg/billing/subscription_test.go]: This may not work after Go 1.8.

File: pkg/billing/subscription_test.go [hunk 2]
```diff
@@ -101,17 +103,173 @@ func (s *SubscriptionTestSuite) TestSubscriptionBuy() {
...
+
+	resp, err := new(SubscriptionServer).Buy(s.customer1.context, &proto.SubscriptionBuyRequest{
+		ProductUid: "BASIC-M2M",
+		CycleCount: 12,
+	})
+	s.Assert().NoError(err)
+	log.Println(*resp.Status)
+	s.Assert().True(resp.Status.IsOK())
+
+	listResp, err := new(SubscriptionServer).List(s.customer1.context, &proto.SubscriptionListRequest{})
```
Comment (TamalSaha) [inline pkg/billing/subscription_test.go]: Why not just create the server once?

File: pkg/billing/subscription_test.go (hunk 1 above)
Comment (sadlil) [inline pkg/billing/subscription_test.go]: As far as i know they are setting a default place if `$GOPATH` is not set.
https://go-review.googlesource.com/c/33730/5/src/cmd/go/alldocs.go#932

There may be a function That can handle this.

File: pkg/billing/subscription_test.go (hunk 2 above)
Comment (sadlil) [inline pkg/billing/subscription_test.go]: There is no specific reason. Just this one looked better. This also doesn't do any performance lagging. Should i change?

File: pkg/billing/billingsuite_test.go [hunk 3]
```diff
@@ -150,3 +156,22 @@ func (s *billingSuite) initBackgroundContext() {
 	)
 	s.customer2.auth = auths
 }
+
+func (s *billingSuite) setupTestInsertProduct() {
+	dataBytes, err := ioutil.ReadFile(os.ExpandEnv("$GOPATH") + "/src/github.com/cowrypay/dealer/hack/testdata/per-seat.json")
```
Comment (TamalSaha) [inline pkg/billing/billingsuite_test.go]: Use https://github.com/appscode/go/blob/master/runtime/runtime.go#L132

**Comment by TamalSaha** [comment]
@sadlil https://github.com/cowrypay/dealer/pull/170

For lease start, either the reserve fails, or you must provide an `ItemUid`. Without an `ItemUid`, a successful lease can't be expired.

**Pull request merged**